### ISSUE TITLE
TagBot experienced an unexpected internal failure, AssertionError: unsupported encoding: none

### ISSUE THREAD

**Issue by sumiya11** [JuliaRegistries/TagBot#350]
In Groebner.jl, TagBot produces "TagBot experienced an unexpected internal failure" accompanied by the message "AssertionError: unsupported encoding: none":

https://github.com/sumiya11/Groebner.jl/actions/runs/10978120022/job/30480878080#step:3:82

Seemed to work fine previously: https://github.com/sumiya11/Groebner.jl/issues/15#issuecomment-2365452551

Not sure if this is a problem in Groebner or something went wrong in TagBot. Any help is appreciated; thanks !

**Comment by JoshuaLampert** [comment]
We are experiencing the same in https://github.com/mikeingold/MeshIntegrals.jl/actions/runs/10978119797/job/30480877628#step:3:64.

**Comment by JoshuaLampert** [comment]
I checked the [latest package releases](https://github.com/JuliaRegistries/General/pulls?page=1&q=sort%3Aupdated-desc+is%3Apr+is%3Aclosed) and it seems like since [this PR](https://github.com/JuliaRegistries/General/pull/115459) (see https://github.com/nflverse/NFLData.jl/actions/runs/10976230101/job/30476748926#step:3:64) TagBot failed every time.

**Comment by IanButterworth** [comment]
What AI thinks the issue could be
```
1) Incorrect File Encoding on GitHub:

The file TagBot is trying to access might not be base64 encoded. This could happen if the file is empty, binary, or improperly formatted.

2) GitHub API Changes:

There might have been changes in the GitHub API that affect how content encoding is handled, causing unexpected values like "none" to appear.
TagBot or Dependency Issues:

3) The version of TagBot or its dependencies (like the GitHub library) might have bugs or incompatibilities leading to this error.
Network or Fetching Issues:

4) Temporary issues when fetching the file could result in incomplete or improperly encoded data.
```

I just updated our deps to latest https://github.com/JuliaRegistries/TagBot/releases/tag/v1.19.5

**Comment by IanButterworth** [comment]
@simeonschaub reported it's still happening on that new release https://github.com/JuliaLabs/AztecDiamonds.jl/actions/runs/10981139085/job/30488183323#step:2:1

**Comment by IanButterworth** [comment]
It may be informative that CI here is passing, but this is failing in the wild, if we can figure out what the difference is.

**Comment by IanButterworth** [comment]
hmmm.. this smells like it might be taking testing off into a different state than normal use
https://github.com/JuliaRegistries/TagBot/blob/4820024f1ddccf5e4e8832f01ecf2a79f62602eb/tagbot/action/repo.py#L88-L94

**Comment by simeonschaub** [comment]
I believe we're running into issues because `Registry.toml` in General has become larger than 1MB:

```python
In [24]: from github import Github, Auth

In [25]: auth = Auth.Token(token)

In [26]: g = Github(auth=auth)

In [27]: repo = g.get_repo("JuliaRegistries/General")

In [28]: registry_toml = repo.get_contents("Registry.toml")

In [29]: registry_toml.encoding
Out[29]: 'none'
```

See https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content:

> If the requested file's size is:
>
>    * 1 MB or smaller: All features of this endpoint are supported.
>    * Between 1-100 MB: Only the raw or object custom media types are supported. Both will work as normal, except that when using the object media type, the content field will be an empty string and the encoding field will be "none". To get the contents of these larger files, use the raw media type.

**Comment by JoshuaLampert** [comment]
Yes, that makes very much sense! I checked that before [this commit](https://github.com/JuliaRegistries/General/commit/c5ab081c2d9ace4ac1c8ca7ac6bf429c2a92c09e) the size of [Registry.toml](https://github.com/JuliaRegistries/General/blob/c5ab081c2d9ace4ac1c8ca7ac6bf429c2a92c09e/Registry.toml) was below 1MB and [after that](https://github.com/JuliaRegistries/General/blob/502cfeee586724062613ec9875b5cccaf8a1af93/Registry.toml) it is above 1MB. This is the same commit, where TagBot started to fail.

**Comment by IanButterworth** [comment]
Thanks. New patch release just cut with that fix. Please confirm it fixes it.

**Comment by JoshuaLampert** [comment]
Seems to work: https://github.com/mikeingold/MeshIntegrals.jl/actions/runs/10993022532/job/30518618465. Thanks q lot @simeonschaub and @IanButterworth!

**Comment by simeonschaub** [comment]
Can confirm as well! 🎉 https://github.com/JuliaLabs/AztecDiamonds.jl/actions/runs/10990108308/job/30518644915

**Comment by jkrumbiegel** [comment]
For me as well https://github.com/PumasAI/SummaryTables.jl/actions/runs/10992929480/job/30518753493
